addTourpackageProcess.php

```diff
@@ -0,0 +1,167 @@
+$dataObject = json_decode($jsonText);
+
+$name = $dataObject->name;
+$price = $dataObject->price;
+$description = $dataObject->description;
+$destinationList = explode(",", $dataObject->destinationList);
+$noOfVehicles = $dataObject->nuOfVehicles;
+$vehicleList = explode(",", $dataObject->vehicleList);
+$hotelList = explode(",", $dataObject->hotelList);
+$duration = $dataObject->duration;
+$milage = $dataObject->milage;
+$activityType = $dataObject->activityType;
+$imageFile = null;
+$mainImageType = null;
+
+if (empty($name)) {
+    echo "Please enter a name for the package";
+} else if (strlen($name) > 45) {
+    echo "Name is too long";
+} else if (empty($price)) {
+    echo "Enter the package price";
+} else if (empty($description)) {
+    echo "Enter the details of the package";
+} else if (strlen($description) > 10000) {
+    echo "Description is too long";
+} else if (empty($destinationList)) {
+    echo "Please select the destinations";
+} else if (empty($noOfVehicles)) {
+    echo "Please enter the number of vehicles alligned for the package";
+} else if (empty($vehicleList)) {
+    echo "Plese select the vehicles";
+} else if ($noOfVehicles != sizeof($vehicleList)) {
+    echo "Please check the number of vehicles and selected vehicles";
+} else if (empty($hotelList)) {
+    echo "Please select the hotels";
+} else if (empty($duration)) {
+    echo "Select the duration of the tour";
+} else if (empty($milage)) {
+    echo "Please enter the milage of the tour";
+} else if (empty($activityType)) {
+    echo "Select the activity type(s)";
+} else {
+    // get vehicle results from the vehicle table according to the vehicle id
+    for ($x = 0; $x < sizeof($vehicleList); $x++) {
+        $vehicleResultSet = Database::search("SELECT * FROM `vehicle` WHERE `id`='" . $vehicleList[$x] . "'");
+        $vehicleNumRows = $vehicleResultSet->num_rows;
+        if ($vehicleNumRows == 1) {
+            $vehicleData = $vehicleResultSet->fetch_assoc();
+
+            // check the vehicles are assigned for any tour package
+            $vtResultSet = Database::search("SELECT * FROM `vehicle_has_tour_package` WHERE `vehicle_id`='" . $vehicleList[$x] . "'");
+            $vtNumRows = $vtResultSet->num_rows;
+            if ($vtNumRows > 0) {
+                echo $vehicleData["number"] . "has already assigned with some package";
+                exit();
+            }
+        } else {
+            echo "No results found for vehicle";
+            exit();
+        }
+    }
+
+    // insert data to the tour_package table
+    Database::insertUpdateDelete("INSERT INTO `tour_package` (`name`, `price`, `description`, `duration`, `total_milage`, `no_of_vehicles`) 
+    VALUES ('" . $name . "', '" . $price . "', '" . $description . "', '" . $duration . "', '" . $milage . "', '" . $noOfVehicles . "')");
+
+    // get lastly inserted record
+    $tpResultSet = Database::search("SELECT * FROM `tour_package` ORDER BY `id` DESC LIMIT 1");
```

Comment:
This method is incorrect. This method is very dangerous method for get newly added `tour_package`. The reason is this all processes are simultaneously executed on the server per request. Then it can be get wrong `tour_package` resultset. This happens because your PHP web application doesn't have `concurrent programming` facility. To fix this, use a another full search using old `tour_package` data.

Reply:
OK